src/models/operation-trait.ts

```diff
@@ -1,3 +1,14 @@
 import type { BaseModel } from "./base";
+import type { BindingsMixinInterface, DescriptionMixinInterface, ExtensionsMixinInterface, ExternalDocumentationMixinInterface, TagsMixinInterface } from './mixins';
+import type { OperationKind } from "./operation";
+import type { SecurityRequirementsInterface } from "./security-requirements";
 
-export interface OperationTraitInterface extends BaseModel {}
+export interface OperationTraitInterface extends BaseModel, BindingsMixinInterface, DescriptionMixinInterface, ExtensionsMixinInterface, ExternalDocumentationMixinInterface, TagsMixinInterface {
+  id(): string;
+  kind(): OperationKind;
```

Comment:
This should be added into the Parser-API https://github.com/asyncapi/parser-api/blob/master/docs/v1.md#operation, as well as removing some of the methods related to Publish/Subscribe

Reply:
Also, `kind` VS `type`. For example, in Security Requirements, we use `type` https://github.com/asyncapi/parser-js/pull/542/files#diff-dfe2dcce2b8c529ef868faf21548e2ba16163b0c4b9ef33129ac286346f2e008R15

Reply:
You're right, we have inconsistencies. `type` in SecurityRequirements and here `kind`. In 3.0.0 we have for example called it as `action` (https://github.com/asyncapi/spec/issues/618) so maybe we should call it `action`?

Reply:
Indeed, `action` sounds like a better fit.